File: cef-web/src/main/java/gov/epa/cef/web/service/EmissionFactorService.java
/*
 * © Copyright 2019 EPA CAERS Project Team
 *
 * This file is part of the Common Air Emissions Reporting System (CAERS).
 *
 * CAERS is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * CAERS is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
 * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with CAERS.  If
 * not, see <https://www.gnu.org/licenses/>.
*/
package gov.epa.cef.web.service;

import java.util.List;

import gov.epa.cef.web.service.dto.EmissionFactorDto;
import gov.epa.cef.web.service.dto.EmissionFormulaVariableCodeDto;

public interface EmissionFactorService {

    /**
     * Search for Emission Factors matching the provided criteria
     * @param dto
     * @return
     */
    List<EmissionFactorDto> retrieveByExample(EmissionFactorDto dto);

    List<EmissionFactorDto> retrieveSLTEfByExample(EmissionFactorDto dto);

    /**
     * Generate variables for an emission factor formula
     * @param formula
     * @return
     */
    List<EmissionFormulaVariableCodeDto> parseFormulaVariables(String formula);

    EmissionFactorDto retrieveByWebfireId(Long webfireId);

    /**
     * Retrieve the Emission Factor with the associated GHG ID
     * @param ghgId
     * @return
     */
    EmissionFactorDto retrieveByGhgId(Long ghgId);

}

File: cef-web/src/main/webapp/src/app/core/services/emission-factor.service.ts
/*
 * © Copyright 2019 EPA CAERS Project Team
 *
 * This file is part of the Common Air Emissions Reporting System (CAERS).
 *
 * CAERS is free software: you can redistribute it and/or modify it under the
 * terms of the GNU General Public License as published by the Free Software Foundation,
 * either version 3 of the License, or (at your option) any later version.
 *
 * CAERS is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
 * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License along with CAERS.  If
 * not, see <https://www.gnu.org/licenses/>.
*/
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { EmissionFactor } from 'src/app/shared/models/emission-factor';

@Injectable({
  providedIn: 'root'
})
export class EmissionFactorService {

  private baseUrl = 'api/emissionFactor';  // URL to web api

  constructor(private http: HttpClient) { }

  search(criteria: EmissionFactor, slt?: boolean): Observable<EmissionFactor[]> {
    let url = `${this.baseUrl}`;

    if (slt) {
        url += '/slt'
    }

    return this.http.post<EmissionFactor[]>(url, criteria);
  }

  retrieveByWebfireId(webfireId: number): Observable<EmissionFactor> {
    const url = `${this.baseUrl}/webfireId/${webfireId}`;
    return this.http.get<EmissionFactor>(url);
  }

  retrieveByGhgId(ghgId: number): Observable<EmissionFactor> {
      const url = `${this.baseUrl}/ghgId/${ghgId}`;
      return this.http.get<EmissionFactor>(url);
  }
}
